### DISCUSSION TITLE
any way to get currentSlide in script?

### DISCUSSION THREAD

**Discussion by dullmode** [ismail9k/vue3-carousel#118] [Q&A] [answered]
I want to get currentSlide in `<script></script>`.

I know there is way below in template

`  <template #addons="{ currentSlide }">`

is there any way to get currentSlide in script?

**Comment by ismail9k** [comment, marked as answer]
You can pass a variable to the carousel via the `v-model` prop, it will be updated with the current slide value. Or you can add a ref to the carousel component and you will have access to all data and methods.

**Reply by dullmode** [reply]
Thank you! I finally solved!!😀

**Reply by tobelyanr** [reply]
i'm kind of new in VUE, can you please drop code piece where you are getting currentSlide index in script?

**Reply by ismail9k** [reply]
```vue
<template>
  <Carousel v-model="currentSlide">
    <Slide v-for="slide in 10" :key="slide">
      <div class="carousel__item">{{ slide }}</div>
    </Slide>

    <template #addons>
      <Navigation />
      <Pagination />
    </template>
  </Carousel>
</template>

<script>
import { defineComponent } from 'vue';
import { Carousel, Pagination, Navigation, Slide } from 'vue3-carousel';

import 'vue3-carousel/carousel.css';

export default defineComponent({
  name: 'Basic',
  data: () => ({ 
     currentSlide: 0,
  }),
  components: {
    Carousel,
    Slide,
    Pagination,
    Navigation,
  },
});
</script>
```
